clustersConfig.py

```diff
@@ -230,6 +245,17 @@ def __init__(
         for c in cc["postconfig"]:
             self.postconfig.append(ExtraConfigArgs(**c))
 
+        if test_only:
+            # Skip the remaining steps. They access the system, which makes them
+            # unsuitable for unit tests.
+            #
+            # TODO: this flag will go away, and instead the test can inject the pieces
+            # that are needed.
+            return
+
+        if self.kind == "openshift":
+            self.configure_ip_range()
+
```

Comment:
This points out an issue with this function. The fact that there are steps in this function that also access the system indicates that this function is doing too much. Can we split it up into two functions and test only the first?

Reply:
I replied to this [here](https://github.com/bn222/cluster-deployment-automation/pull/239#discussion_r1742566553).

This will go away, the patch for that is already done on `th/clustersConfig` branch. But things change towards that goal in certain small steps. In this case, add the unit tests first, then rework the code bit-by-bit to get there. That order of steps calls for the "test_only" flag for the moment.

Marking as resolved, please follow up [here](https://github.com/bn222/cluster-deployment-automation/pull/239#discussion_r1742566553).